### PR TITLE
feat(SpEL): implement to configure the limit of characters for SpEL expressions

### PR DESCRIPTION
**Pull request by j-sandy**
Spring Expression Lanuage (SpEL) has a default limit of 10,000 characters. Springframework provides the feature to configure the limit. This feature allows to configure the limit of characters for SpEL expressions.

Approach:
In order to use an expression with characters more than the given default limit, require to follow either of the below approaches:
1. For Springframework >=5.3.28 and <6.1.3, by setting `maximumExpressionLength` field while instantiating the custom `SpelParserConfiguration` class. 
https://github.com/spring-projects/spring-framework/issues/30380 
https://github.com/spring-projects/spring-framework/issues/30446
2. For Springframework >=6.1.3, by setting a JVM system property or Spring property named `spring.context.expression.maxLength` to the maximum expression length needed by your application. 
https://github.com/spring-projects/spring-framework/issues/31952 
https://github.com/spring-projects/spring-framework/commit/785598629abda944343a02307ad82a79bb31b589 

Spinnaker supports spring boot 2.7.18, that brings springframework 5.3.31 [https://docs.spring.io/spring-boot/docs/2.7.18/reference/html/dependency-versions.html#appendix.dependency-versions.propertie9]. So first approach need to be implemented along with spinnaker enhancement to expose the `maximumExpressionLength` field.

### CODE REVIEW

**Review by dbyron-sf — CHANGES_REQUESTED** [review]

File: kork-expressions/src/main/java/com/netflix/spinnaker/kork/expressions/config/ExpressionProperties.java [hunk 1]
```diff
@@ -26,6 +27,9 @@ public class ExpressionProperties {
 
   private final FeatureFlag doNotEvalSpel = new FeatureFlag().setEnabled(true);
 
+  @Value("${expression.max-expression-length:-1}")
+  private int maxExpressionLength;
```
[the remark this answers is not in the document]
Comment (j-sandy) [reply kork-expressions/src/main/java/com/netflix/spinnaker/kork/expressions/config/ExpressionProperties.java]: Updated the code.

File: kork-expressions/src/main/java/com/netflix/spinnaker/kork/expressions/config/ExpressionProperties.java [hunk 2]
```diff
@@ -26,6 +26,8 @@ public class ExpressionProperties {
 
   private final FeatureFlag doNotEvalSpel = new FeatureFlag().setEnabled(true);
 
+  private int maxExpressionLength = -1;
```
Comment (dbyron-sf) [inline kork-expressions/src/main/java/com/netflix/spinnaker/kork/expressions/config/ExpressionProperties.java:29]: javadoc please.

File: kork-expressions/src/main/java/com/netflix/spinnaker/kork/expressions/config/ExpressionProperties.java [hunk 3]
```diff
@@ -20,12 +20,17 @@
...
 
+/** Configuration properties for Spring Expression Language (SpEL) evaluation related features. */
 @Data
 @ConfigurationProperties(prefix = "expression")
 public class ExpressionProperties {
 
+  /** Flag to determine if SpEL evaluation to be skipped. */
   private final FeatureFlag doNotEvalSpel = new FeatureFlag().setEnabled(true);
 
+  /** To set the maximum limit of characters in expression for SpEL evaluation. */
```
Comment (dbyron-sf) [inline kork-expressions/src/main/java/com/netflix/spinnaker/kork/expressions/config/ExpressionProperties.java:31]: Please mention something about what -1 means.  I think it means use the limit from spring (e.g. 10000), as opposed to no limit.

**Review by dbyron-sf — APPROVED** [review]

**Comment by dbyron-sf** [comment]
@mergifyio update

**Pull request merged**